### DISCUSSION TITLE
[Shape Error] UNETR patch size and image size mismatch

### DISCUSSION THREAD

**Discussion by aalhayali** [Project-MONAI/MONAI#5359] [Q&A]
Hi everyone!

I have a quick question regarding an error I'm facing and I'm a bit confused on how to pursue it.

I am working on sagittal MRI images of size (512,512,7) (H,W,Slices). I applied the `Resized` transform and resized the image to (7,256,256), along with other transformations. The resultant original image are as follows:

![download](https://user-images.githubusercontent.com/65701637/196597819-ddc5f4b6-fe90-4d65-8f26-47e7b76163ff.png)

The sizes for the original and transformed images are (2, 1, 7, 512, 512), and (2, 1, 7, 256, 256) respectively. When building the UNETR model, I used (7,256,256) model parameter and I'm getting the following error:

`ValueError: patch_size should be smaller than img_size.`

I am unsure why this error is being output although I am inputting the input image size generated from the `Resized` transform, as per the documentation _img_size (Union[Sequence[int], int]) – dimension of input image._

I would appreciate any suggestions or pointers to fix that error!

model code:
`
model = UNETR(
    in_channels=1, 
    out_channels=2, 
    img_size=(7,256,256), 
    feature_size=16, 
    norm_name='batch', 
    spatial_dims=3,
    ).to(device)
`

**Comment by Nic-Ma** [comment]
Hi @tangy5 ,

Could you please help take a look at this problem?

Thanks in advance.

**Comment by tangy5** [comment]
@aalhayali Hi, thanks for the question. Great to see UNETR is used for broader datasets. 
From the description, I found your data is a bit special, there are only 7 slices. However, the default patch size of 3D transformer in UNETR is (16, 16, 16), which means the dimension at each axis should be at least 16.  I have some suggestions:
1. You could try a 2D model for training your task, using slices as inputs. 
2. Upsample, in slice thickness, make the slices numbers to 32 for training a 3D model.  (add Spacingd transform)
Thanks.

**Reply by aalhayali** [reply]
Thanks!

I will try the 2D approach as a baseline model, then will try to upscale it. My images have poor quality given the number of slice, with a wide FOV. I would ideally rather avoid step 2, but I will try it out for the sake of experimentation. May I know why the slice number you chose is 32?
